src/openforms/forms/api/validators.py

```diff
@@ -100,10 +100,10 @@ def _is_advanced_logic(
                 return True
         return False
 
-    def _is_date_operand(self, operand: Any) -> bool:
+    def _is_date_or_datetime_operand(self, operand: Any) -> bool:
         return (
             isinstance(operand, Operation)
-            and operand.operator == "date"
+            and (operand.operator == "date" or operand.operator == "datetime")
```

Comment:
neater would be `operand.operator in ["date", "datetime"]`, but good enough for now